core/src/test/java/org/apache/iceberg/rest/TestRESTCatalog.java

```diff
@@ -2409,7 +2409,7 @@ public void testPaginationForListTables() {
     RESTCatalog catalog =
         new RESTCatalog(SessionCatalog.SessionContext.createEmpty(), (config) -> adapter);
     catalog.initialize("test", ImmutableMap.of(RESTSessionCatalog.REST_PAGE_SIZE, "10"));
-    int numberOfItems = 30;
+    int numberOfItems = 28;
```

Comment:
[minor] can we still keep the test with number of items as multiples of page size

Reply:
I think maybe there may be some case @rcjverhoef is trying to test where the page size is larger than the remaining items left in the result set. If so maybe it might be better to keep the original `numberOfItems` for this test and can add another test for this specific logic you are want to hit?

Reply:
+1 wanted both to co-exist !

Reply:
Same comment as below: The test case will work if and only if `numberOfItems` is a multiple of the requested page size. Any test where that is not the case and we request the last page will break in an `IndexOutOfBoundsException`. 

So I first changed this to consistently have the error, then fixed it

Reply:
There is something to be said for having the `pageSize` be a multiple of the `numberOfItems` since that is an interesting edge case to ensure the return `nextPageToken` is empty from that point forward. Let me see if I can redo the tests somewhat to cover both cases.

Reply:
I pushed a simple change make the test parameterized and test a few cases: 21, 25, 29, 30. 

- 21 is an edge case since that is the first amount we have 3 pages to present with the test setup
- 25 & 29 are somewhat cases when we are half or 90% of the page but nothing after
- 30 which is a page full, but like the others should return an empty page-token

25 & 29 might be some superfluous to be honest. Might also make an argument for testing 21-30 entirely then.

Thoughts?

Reply:
Thanks @rcjverhoef for the fix, as well as for the detailed explanation. i think I agree that the main two test cases to try would be if its an exact multiple and if it has some remainder. I think numbers between 21- 30 personally is not needed. So to me having `21` alone and `30` alone should suffice. 

cc @singhpk234 @amogh-jahagirdar

Reply:
+1 Imho the two cases should be suffice to cover 100% of the lines

Reply:
sounds good to me! Updated the parameterized test to only do 21 & 30.

Reply:
Thankyou @rcjverhoef !